### ISSUE TITLE
git CA config doesn't work on Debian 7

### ISSUE THREAD

**Issue by pinkencryption** [flynn/flynn#3204]
I installed Flynn CLI on another machine
Added the key and I am able to use flynn apps to be created and deleted
However my git push are being rejected.

Error I am recieving is as below ::

error: server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none while accessing https://git.585.io/shootthemup2.git/info/refs
fatal: HTTP request failed

Please help me out.

**Comment by lmars** [comment]
@pinkencryption did you run the `flynn cluster add ...` command (from the bootstrap output) on the second machine? You need to run that so that it can configure `git` with the cluster CA certificate.

**Comment by pinkencryption** [comment]
Hi, yes I did run the command, Deleted cluster, run it again. I had to now disable certificate verification but it is not advisable

**Comment by titanous** [comment]
Do you see an entry in `~/.gitconfig` specifying the CA certificate and credential helper? It should look like this:

```text
[http "https://git.o5ol.flynnhub.com"]
	sslCAInfo = /Users/titanous/.flynn/ca-certs/flynn-1466789121.pem
[credential "https://git.o5ol.flynnhub.com"]
	helper = /usr/local/bin/flynn git-credentials
```

**Comment by titanous** [comment]
Is this the same problem as #3214?

**Comment by pinkencryption** [comment]
No. this is a different problem all together, this is linux. That one is windows.

**Comment by titanous** [comment]
Okay, as I mentioned, please post the contents of ~/.gitconfig.

**Comment by pinkencryption** [comment]
root@myvm:/# cat ~/.gitconfig
[http "https://git.585.io"]
	sslCAInfo = /root/.flynn/ca-certs/default.pem
[credential "https://git.585.io"]
	helper = /usr/local/bin/flynn git-credentials

**Comment by titanous** [comment]
Does `/root/.flynn/ca-certs/default.pem` exist and is it readable? What version of git are you using?

**Comment by pinkencryption** [comment]
ls -ltrha /root/.flynn/ca-certs/default.pem
-rw-r--r-- 1 root root 1.1K Jul 30 01:30 /root/.flynn/ca-certs/default.pem

**Comment by pinkencryption** [comment]
git --version
git version 1.7.10.4

**Comment by titanous** [comment]
Please run this command and provide the full output:

```text
GIT_TRACE=1 GIT_CURL_VERBOSE=1 git push -v flynn master
```

**Comment by pinkencryption** [comment]
http://pastebin.com/T33KXAyB

**Comment by titanous** [comment]
Looks like it's fine? Do you have a flag set somewhere to disable certificate verification? If so, get rid of that and run it again.

**Comment by pinkencryption** [comment]
http://pastebin.com/Uj3LizK5/?e=1

**Comment by titanous** [comment]
What user are you running the `git push` as?

**Comment by pinkencryption** [comment]
User : root

**Comment by titanous** [comment]
Is there a `CAfile` section in `~/.gitconfig`? If so, what is the section header?

**Comment by pinkencryption** [comment]
cat ~/.gitconfig
[http "https://git.585.io"]
	sslCAInfo = /root/.flynn/ca-certs/default.pem
[credential "https://git.585.io"]
	helper = /usr/local/bin/flynn git-credentials

**Comment by pinkencryption** [comment]
thats all I have there.

**Comment by titanous** [comment]
Okay, thanks, what distro and release?

**Comment by pinkencryption** [comment]
cat /etc/*-release
PRETTY_NAME="Debian GNU/Linux 7 (wheezy)"
NAME="Debian GNU/Linux"
VERSION_ID="7"
VERSION="7 (wheezy)"
ID=debian
ANSI_COLOR="1;31"
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support/"
BUG_REPORT_URL="http://bugs.debian.org/

**Comment by pinkencryption** [comment]
Kernel : 2.6.32-042stab116.1

**Comment by titanous** [comment]
Thanks, we'll look into this. I suspect there is something overriding/ignoring the `sslCAInfo` setting.

**Comment by pinkencryption** [comment]
Okay

**Comment by titanous** [comment]
This is because the default version of git (1.7.10) on Debian wheezy is too old. It can be fixed by installing the backported version (1.9.1):

```text
echo "deb http://ftp.debian.org/debian wheezy-backports main" >> /etc/apt/sources.list
apt-get update
apt-get -t wheezy-backports install git
```
